File: Easy/25- 1688. Count of Matches in Tournament/1688. Count of Matches in Tournament.py
class Solution:
    def numberOfMatches(self, n: int) -> int:
        """
        There are n teams. Each match results in the elimination of
        one team. After n - 1 matches, there is one team left standing;
        it becomes the winner.
	    """
        return n - 1
    
    
    """
    n=14 --> output=13
    n=7  --> output=6
    """

File: Easy/25- 1688. Count of Matches in Tournament/1688. Count of Matches in Tournament.java
class Solution {
    public int numberOfMatches(int n) {
        int numberOfMatches=0;
        
        while(n!=1){
            if(n%2==0){
                numberOfMatches+=(n/2);
                n=n/2;
            }
            else if(n%2==1){
                numberOfMatches+=((n-1)/2);
                n=((n-1)/2)+1;
            }
        }
        
        return numberOfMatches;
    }
}